src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c

```diff
@@ -26,25 +48,312 @@ const ActorInit En_Wonder_Item_InitVars = {
+    this->actor.unk_1F = 1;
+    if ((this->switchFlag >= 0) && Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ Ｙｏｕ ａｒｅ Ｓｈｏｃｋ！  ☆☆☆☆☆ %d\n" VT_RST, this->switchFlag);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    switch (this->wonderMode) {
+        case WONDERITEM_MULTITAG_FREE:
+            this->numTagPoints = this->actor.posRot.rot.z & 0xF;
+            rotZover10 = 0;
+            if (this->actor.posRot.rot.z >= 10) {
+                rotZover10 = this->actor.posRot.rot.z / 10;
+                this->timerMod = rotZover10 * 20;
+            }
+            this->numTagPoints = this->actor.posRot.rot.z - rotZover10 * 10;
+            // i.e timerMod = rot.z / 10 seconds, numTagPoints = rot.z % 10
+            this->updateFunc = EnWonderItem_MultitagFree;
+            break;
+        case WONDERITEM_TAG_POINT_FREE:
+            tagIndex = this->actor.posRot.rot.z & 0xFF;
+            sTagPointsFree[tagIndex] = this->actor.posRot.pos;
+            Actor_Kill(&this->actor);
+            break;
+        case WONDERITEM_PROXIMITY_DROP:
+            this->dropCount = this->actor.posRot.rot.z & 0xFF;
+            this->updateFunc = EnWonderItem_ProximityDrop;
+            break;
+        case WONDERITEM_INTERACT_SWITCH:
+            colTypeIndex = this->actor.posRot.rot.z & 0xFF;
+            Collider_InitCylinder(globalCtx, &this->collider);
+            Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+            this->collider.body.bumper.flags = collisionTypes[colTypeIndex];
+            this->collider.dim.radius = 20;
+            this->collider.dim.height = 30;
+            this->updateFunc = EnWonderItem_InteractSwitch;
+            break;
+        case WONDERITEM_UNUSED:
+            break;
+        case WONDERITEM_MULTITAG_ORDERED:
+            this->numTagPoints = this->actor.posRot.rot.z & 0xF;
+            rotZover10 = 0;
+            if (this->actor.posRot.rot.z >= 10) {
+                rotZover10 = this->actor.posRot.rot.z / 10;
+                this->timerMod = rotZover10 * 20;
+            }
+            this->numTagPoints = this->actor.posRot.rot.z - rotZover10 * 10;
+            // i.e timerMod = rot.z / 10 seconds, numTagPoints = rot.z % 10
+            this->updateFunc = EnWonderItem_MultitagOrdered;
+            break;
+        case WONDERITEM_TAG_POINT_ORDERED:
+            tagIndex = this->actor.posRot.rot.z & 0xFF;
+            sTagPointsOrdered[tagIndex] = this->actor.posRot.pos;
+            Actor_Kill(&this->actor);
+            break;
+        case WONDERITEM_PROXIMITY_SWITCH:
+            this->updateFunc = EnWonderItem_ProximitySwitch;
+            break;
+        case WONDERITEM_BOMB_SOLDIER:
+            Collider_InitCylinder(globalCtx, &this->collider);
+            Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+            this->collider.body.bumper.flags = WONDERITEM_DAMAGE_SLINGSHOT;
```

Comment:
wont these defines be provided by the collision check system? dont think we should make actor specific ones

Reply:
Not for a while. I think we'll need a lot more actors decomped before it becomes clear which dFlag combinations will want macros. In the meantime, I don't see any harm in including these for readability. They can be easily replaced later.

Reply:
people often refer to other files to see what conventions we currently have, and may see this and think making their own defines is required. i would prefer a comment that is removed later honestly